components/automate-ui/src/app/entities/environments/environment.model.ts

```diff
@@ -1,3 +1,41 @@
 export interface Environment {
   name: string;
+  chef_type: string;
+  description: string;
+  json_class: string;
+  default_attributes: string;
+  override_attributes: string;
+  run_list: string[];
+  cookbook_versions: CookbookVersion[];
+}
+
+export interface CookbookVersion {
+  name: string;
+  version: string;
+}
+
+export class EnvironmentAttributes {
+  default_attributes: Object;
+  override_attributes: Object;
+  all: AllAttributes;
+  constructor(resp: RespEnvironmentAttributes) {
+    this.default_attributes =
+      (resp.default_attributes && JSON.parse(resp.default_attributes)) || {};
+    this.override_attributes =
+      (resp.override_attributes && JSON.parse(resp.override_attributes)) || {};
+    this.all = {
+      default_attributes: this.default_attributes,
+      override_attributes: this.override_attributes
+    };
+  }
+}
+
+export interface RespEnvironmentAttributes {
+  default_attributes: string;
+  override_attributes: string;
+}
+
+export interface AllAttributes {
+  default_attributes: Object;
+  override_attributes: Object;
```

Comment:
Prefer not to duplicate things (these are both here and infra-role.model.ts).
Suggest:
(1) delete these lines here.
(2) rename `RespRoleAttributes` to `RespAttributes`.
(3) import `RespAttributes` and `AllAttributes` from infra-role.model.ts in this file

Reply:
I have incorporated review comments. :)